lib/vector-core/src/metrics/ddsketch.rs

```diff
@@ -644,7 +649,11 @@ impl AgentDDSketch {
             .or(Some(f64::NAN))
     }
 
-    pub fn merge(&mut self, other: AgentDDSketch) {
+    pub fn merge(&mut self, other: &AgentDDSketch) -> bool {
```

Comment:
`merge` returning a `bool` feels weird to me, since the function name implies an action - not a "check".

Reply:
This is true, and a `Result` would probably better.

Reply:
Addressed in 1021d2416.